docs/source/multiple-threads.rst

```diff
@@ -0,0 +1,150 @@
+
+*Global Synchronization* creates a *happens-before* relationship between the completion of every *Fundamental Operation* on any *Execution Space Instance* that *happens-before* the *Global Synchronization* and the thread that performs the *Global Synchronization*.
+
+.. Should the above actually be *synchronizes-with*?
+
+A multi-threaded program may concurrently perform *Fundamental Operations* on distinct *Execution Space Instances*. The order in which these operations execute is indeterminate.
+
+``View``
+--------
+
+* Managed Construction
+  Managed construction of a Kokkos View performs a *Memory Allocation*, potentially followed by a *Parallel Dispatch* to initialize the memory (depending on whether ``WithoutInitializing`` was passed), potentially followed by a *Synchronization* (if no execution space instance was passed, so that allocation and initialization *happen-before* any subsequent operation that may reference the ``View``'s memory').
+  .. Do we want that to be *Global Synchronization* or *Local Synchronization*?
+  (Note that particular backends may internally perform additional synchronization operations)
+* Unmanaged Construction
+  Unmanaged construction performs no fundamental operations.
+* Destruction
+  Destruction occurs when the reference count of the underlying object to which a ``View`` refers falls to zero. Destruction performs a *Global Synchronization*, followed by a *Memory Deallocation*.
+  The *Global Synchronization* is performed to ensure that any preceding operations that may reference the memory owned by the underlying object *happen-before* that memory is released.
+  Any *Fundamental Operation* which accesses the ``View`` instance performed in a way that it does not have a *happens-before* relationship with Destruction results in undefined behavior.
+  (Note this could occur via improper concurrent use of a ``View`` variable shared between threads, one of which passes it to an operation, and another which reassigns it to reference a different instance)
+* Metadata Query
+* Element Access
+  Element Access performs a Data Access operation.
+
+
+Backend-Specific Details
+------------------------
+
+.. Local or Global synchronizations below?
+
+* ``Serial``
+  This backend is *synchronous* - there is implicitly a *Synchronization* after any *Memory Allocation*, *Memory Deallocation*, *Parallel Dispatch*, or *Data Movement* operation. 
+
+* ``OpenMP``
+  This backend is *synchronous* - there is implicitly a *Synchronization* after any *Memory Allocation*, *Memory Deallocation*, *Parallel Dispatch*, or *Data Movement* operation. 
+
+* ``Threads``
+  This backend is *synchronous* - there is implicitly a *Synchronization* after any *Memory Allocation*, *Memory Deallocation*, *Parallel Dispatch*, or *Data Movement* operation. 
+
+* ``CUDA`` and ``HIP``
+
+* ``HPX``
+
```

Comment:
We should talk more about parallel dispatch and the behavior of independent threads (without a happens-before relationship between them) accessing the same data.
Possibly also clarifying were we promise that dispatch implies fences (linking to API for parallel_for, parallel_reduce, parallel_scan).